### PR TITLE
Nuke gridtilelookup

### PR DESCRIPTION
**Pull request by metalgearsloth**
It's here

### CODE REVIEW

File: Robust.Shared/GameObjects/Systems/EntityLookup.Queries.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        return GetEntitiesIntersecting(tileRef.GridIndex, tileRef.GridIndices);
+    }
+
+    #endregion
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private Box2 GetLocalBounds(Vector2i gridIndices)
+    {
+        return new Box2(gridIndices, gridIndices + 1);
+    }
```
Comment (ElectroJr) [inline Robust.Shared/GameObjects/Systems/EntityLookup.Queries.cs]: Though non-standard grid sizes aren't used anywhere afaik, I guess this should still support them.

**Comment by moonheart08** [comment]
SpawnEntity performance fixing to go up

File: Robust.Shared/GameObjects/Systems/EntityLookup.Queries.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        return GetEntitiesIntersecting(tileRef.GridIndex, tileRef.GridIndices);
+    }
+
+    #endregion
+
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private Box2 GetLocalBounds(Vector2i gridIndices, ushort tileSize)
+    {
+        return new Box2(gridIndices, gridIndices + tileSize);
+    }
```
Comment (ElectroJr) [inline Robust.Shared/GameObjects/Systems/EntityLookup.Queries.cs]:
```suggestion
    {
        return new Box2(gridIndices * tileSize, (gridIndices + 1) * tileSize);
    }
```

Comment (ElectroJr) [reply Robust.Shared/GameObjects/Systems/EntityLookup.Queries.cs]: otherwise looks good.

**Comment by Acruid** [comment]
May resolve https://github.com/space-wizards/RobustToolbox/issues/2377

**Pull request merged**